File: java/src/formula/interval/ListIntervalFormula.java
package formula.interval;

import java.util.ArrayList;
import java.util.Iterator;
import scheme.ListScheme;
import scheme.Interval;


public class ListIntervalFormula implements Iterable<IIntervalFormula> {
	private ArrayList<IIntervalFormula> listFormula;
	
	public ListIntervalFormula() {
		listFormula = new ArrayList<IIntervalFormula>();
	}
	
	public Iterator<IIntervalFormula> iterator() {
		return listFormula.iterator();
	}
	
	private int distance = 0;
	
	public void setDistance(int distance) {
		this.distance = distance;
		
		for (IIntervalFormula f : listFormula) {
			f.setDistance(distance);
		}
	}
	
	public int getDistance() {
		return distance;
	}
	/**
	 * Ajouter une formule (en fin de liste).
	 * @param fml
	 */
	public void add(IIntervalFormula fml) {
		fml.setDistance(distance);
		listFormula.add(fml);		
	}
	
	/**
	 * Ajoute la formule en début de liste.
	 * @param fml
	 */
	public void unshift(IIntervalFormula fml) {
		fml.setDistance(distance);
		listFormula.add(0, fml);		
	}

	public int size() {
		return listFormula.size();
	}

	public IIntervalFormula get(int idx) {
		return listFormula.get(idx);
	}
	
	public void setStats(ListInterval listInterval, ListScheme schemes) {
			
		for (Interval intv : listInterval)
			for (IIntervalFormula iif : listFormula )
				intv.setStat(iif, schemes);
	}

}

File: php/Scheme/Formula/Interval/ListIntervalFormula.php
<?php

namespace Scheme\Formula\Interval;

use Scheme\ListScheme;
use Scheme\Interval;

class ListIntervalFormula {

  private $listFormula;
  private $distance = 0;
	
  public function __construct() {
    $this->listFormula = [];
  }
	
  public function setDistance($distance) {
    $this->distance = $distance;
		
    foreach ($this->listFormula as $key => $formula) {
      $formula->setDistance($this->distance);
    }
  }
	
  public function getDistance() {
    return $this->distance;
  }
  /**
   * Ajouter une formule (en fin de liste).
   * @param fml
   */
  public function add(IIntervalFormula $fml) {
    $fml->setDistance($this->distance);
    $this->listFormula[] = $fml;		
  }
	
  /**
   * Ajoute la formule en début de liste.
   * @param fml
   */
  public function unshift(IIntervalFormula $fml) {
    $fml->setDistance($this->distance);
    unshift($this->listFormula, $fml);		
  }

  public function size() {
    return count($this->listFormula);
  }

  public function get($idx) {
    return $this->listFormula[$idx];
  }
	
  public function setStats(ListInterval $listInterval, ListScheme $schemes) {
			
    /*foreach ($listInterval as $key => $interval) {
      foreach ($this->listFormula as $kkey => $formula) {
	$interval->setStat($formula, $schemes);
      }
      }*/
    $i = 0;
    while ($i < $listInterval->size()) {
      $j = 0;
      while ($j < count($this->listFormula)) {
	$listInterval->get($i)->setStat($this->listFormula[$j], $schemes);
	$j++;
      }
      $i++;
    }

  }

}
